specutils/io/default_loaders/tabular_fits.py

```diff
@@ -74,6 +74,9 @@ def tabular_fits_loader(file_obj, column_mapping=None, hdu=1, **kwargs):
 
     tab = Table.read(file_obj, format='fits', hdu=hdu)
 
+    # Get the header from HDU0
+    tab.meta = fits.getheader(file_obj, 0)
```

Comment:
I think this is breaking compression support since `Table.read()` in those cases closes the `file_obj` on return, so you can no longer read from it. Probably need to get away without using the `table` io at all here as well, i.e. use `read_fileobj_or_hdulist` as `wcs1d_fits_loader` does and read both `tab` and `header` from `hdulist` (not sure if `Table` directly supports this).